### PR TITLE
[Deno] Configure API middleware

### LINKED ISSUE #563: [deno] 1. Configure API middleware
# Background

Regardless of REST or GraphQL, the API will have a middleware that needs configuration for things like CORS or routing for REST and schema stitching for GraphQL. This issue is to setup the API to be usable out of the box.

# Acceptance

- [ ] Update CORS configurations
- [ ] Ensure routing patterns are working and defined
- [ ] Setup any middleware (if applicable)

### PR DESCRIPTION
**Pull request by vyktoremario**
## Type of change

- [x] Configure API middleware

## Summary of change

Closes #563 

## Checklist

- [x] These changes follow the [contributing guidelines](https://github.com/thisdot/starter.dev/blob/main/CONTRIBUTING.md)
- [x] This starter kit has been approved by the maintainers
- [x] Changes for this new starter kit are being pushed to an integration branch instead of main
- [x] All dependencies are version locked
- [ ] This fix resolves #
- [x] I have verified the fix works and introduces no further errors

### CODE REVIEW

File: starters/deno-oak-denodb/.gitignore [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+# vercel
+.vercel
+
+# typescript
+*.tsbuildinfo
```
Comment (dariodjuric) [inline starters/deno-oak-denodb/.gitignore:26]: Could we add something like this for IDEs?

Disclaimer: I stole it from a Stripe app.

```
# IDEs and editors
/.idea
.project
.classpath
.c9/
*.launch
.settings/
*.sublime-workspace
.vscode
```

File: starters/deno-oak-denodb/src/main.ts [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 const router = new Router();
-router.get('/', async ({ request, response }: Context) => {
+router.get('/', ({ request, response }: Context) => {
   response.body = `Hello world! from ${request.url}`;
 });
 
+app.use(
+  oakCors({
+    allowedHeaders: ['Content-Type', 'Authorization'],
+    origin: CORS_ALLOWED_ORIGINS ? CORS_ALLOWED_ORIGINS.split(',') : '',
```
Comment (dariodjuric) [inline starters/deno-oak-denodb/src/main.ts:15]: This is great.

File: starters/deno-oak-denodb/.env.example [hunk 3]
```diff
@@ -0,0 +1,11 @@
+API_HOST_URL='http://localhost:3333'
```
Comment (dariodjuric) [inline starters/deno-oak-denodb/.env.example:1]: It seems that the port information is duplicated in these two lines. What do you think about having something like `API_HOST='http://localhost'` instead? And then you concatanate the two env vars where needed.

**Review by dariodjuric — CHANGES_REQUESTED** [review]
I left some suggestions, otherwise nice job.

File: starters/deno-oak-denodb/.gitignore (hunk 1 above)
Comment (vyktoremario) [reply starters/deno-oak-denodb/.gitignore:26]: Alright sure

File: starters/deno-oak-denodb/src/main.ts (hunk 2 above)
Comment (vyktoremario) [reply starters/deno-oak-denodb/src/main.ts:15]: Thank you🙂

**Review by dariodjuric — APPROVED** [review]

**Pull request merged**